### DISCUSSION TITLE
Rendering code component returns object instead of string.

### DISCUSSION THREAD

**Discussion by emilianoyaryura** [remarkjs/remark#981] [General] [answered]
Hi, how are you? I'm building a blog, and I want to render a custom component for the code blocks:
<img width="575" alt="Screen Shot 2022-04-14 at 14 16 53" src="https://user-images.githubusercontent.com/55447481/163440310-766b7d13-a29e-490c-a072-ea7235cca543.png">

My component consists in the following:
<img width="506" alt="Screen Shot 2022-04-14 at 14 17 33" src="https://user-images.githubusercontent.com/55447481/163440397-8bc70df3-9681-4b74-ab5a-e40d2533e613.png">

But this only renders part of my code (the first object of an array to be precise), but I don't know why it returns an array when its a string...
<img width="847" alt="Screen Shot 2022-04-14 at 14 18 27" src="https://user-images.githubusercontent.com/55447481/163440520-3661d65e-b06d-495b-a8c6-eeb2b9009859.png">
(what it's returning)

<img width="592" alt="Screen Shot 2022-04-14 at 14 20 44" src="https://user-images.githubusercontent.com/55447481/163440849-e395d1bb-7f62-4adc-a758-c6a850815583.png">
(The code string I'm rendering into <ReactMarkdown />)

Thanks!

**Comment by wooorm** [comment, marked as answer]
Please make a repro. Here’s a starter for codesandbox: https://codesandbox.io/s/github/remarkjs/.github/tree/main/sandbox-templates/react-markdown-with-create-react-app. Here for stackblitz: https://stackblitz.com/github/remarkjs/.github/tree/main/sandbox-templates/react-markdown-with-create-react-app
